llvm/lib/IR/RepoHashCalculator.cpp

```diff
@@ -322,16 +322,14 @@ void HashCalculator::hashValue(const Value *V) {
   hashNumber(SN.first->second);
 }
 
-static bool shouldAddGODependency(const GlobalObject &GO) {
+static bool shouldAddGODependency(const Function &Fn) {
```

Comment:
I prefer the original version of this function. Changing the signature to accept `Function` rather than `GlobalObject` embeds the idea that only functions can be dependencies into the function signature. That knowledge should be the province of this function rather its callers. I can’t see any particular advantage in this and whilst the assertion may be true today, it may change in future.

Reply:
Revert back in https://github.com/SNSystems/llvm-project-prepo/pull/127/commits/4a8756041970f24b02df38128f15c683126a7ebe.